frostsnap_coordinator/src/bitcoin.rs

```diff
@@ -3,38 +3,37 @@ pub mod wallet;
 mod wallet_persist;
 
 use bdk_chain::{
-    bitcoin::{
-        self,
-        bip32::{ChildNumber, DerivationPath},
-        key::Secp256k1,
-    },
+    bitcoin::{self, key::Secp256k1},
     miniscript::{
         descriptor::{DerivPaths, DescriptorMultiXKey, Wildcard},
         Descriptor, DescriptorPublicKey,
     },
 };
 use frostsnap_core::{
-    schnorr_fun::fun::Point,
-    tweak::{Account, TweakableKey},
+    tweak::{AppTweakKind, BitcoinAccount, Keychain},
+    MasterAppkey,
 };
 
 pub fn multi_x_descriptor_for_account(
-    root_key: Point,
-    account: Account,
-    network: bitcoin::Network,
+    master_appkey: MasterAppkey,
+    account: BitcoinAccount,
+    network: bitcoin::NetworkKind,
 ) -> Descriptor<DescriptorPublicKey> {
-    let root_bitcoin_xpub = root_key.bitcoin_app_xpub().xpub(network);
-    let account_xpub = root_bitcoin_xpub
-        .derive_pub(&Secp256k1::verification_only(), &account.derivation_path())
+    let secp = Secp256k1::verification_only();
+    let bitcoin_app_xpub = master_appkey.derive_appkey(&secp, AppTweakKind::Bitcoin, network);
+    let account_xpub = bitcoin_app_xpub
+        .derive_pub(&secp, &account.derivation_path())
         .unwrap();
 
+    let keychains = [Keychain::External, Keychain::Internal];
+
     let multi_xpub = DescriptorPublicKey::MultiXPub(DescriptorMultiXKey {
-        origin: Some((root_bitcoin_xpub.fingerprint(), account.derivation_path())),
+        origin: Some((bitcoin_app_xpub.fingerprint(), account.derivation_path())),
         xkey: account_xpub,
         derivation_paths: DerivPaths::new(
-            [0, 1]
+            keychains
                 .into_iter()
-                .map(|i| DerivationPath::from(vec![ChildNumber::Normal { index: i }]))
+                .map(|keychain| keychain.derivation_path())
```

Comment:
I think this is wrong. This path should be the full path from the Bitcoin appkey not just the keychain index

Reply:
(actually it's right the keychain bit is just the bit on the end like `xpub.../<0;1>/*` so it shouldn't have more things in it).